projects/components/src/table/table.component.ts

```diff
@@ -421,10 +421,18 @@ export class TableComponent
     this.changeDetector.markForCheck();
   }
 
-  public toggleRowSelected(row: StatefulTableRow): void {
+  public toggleExistingOrSelectNewRow(row: StatefulTableRow): void {
+    const rowIndexInSelections = (this.selections ?? []).findIndex(selection => isEqualIgnoreFunctions(selection, row));
+    rowIndexInSelections >= 0 ? (this.selections = []) : (this.selections = [row]);
```

Comment:
It's not clear to me how this is different than the multi case? Even if they are slightly different, why do we need two functions?

Reply:
In Multi, if you click on second row then selections array will have two rows. In Single selection, if you click on a new row, the previous row should be unselected or toggled and only the current row should now be selected. Created another method to keep logic separate.

Reply:
Got it - but rather than fork the functions externally, I think it would be much clearer to keep the logic internal to the call so callers don't have to make the determination of the selection mode before deciding which to call. I killed the mutation too because I'm a freak, but I find this much more readable. Probably because everyone finds their own code more readable, so up to you.
```typescript
  public toggleRowSelected(toggledRow: StatefulTableRow): void {
    const previousSelections = this.selections ?? [];
    const deselectedRow = previousSelections.find(selection => isEqualIgnoreFunctions(selection, toggledRow));
    if (deselectedRow) {
      this.selections = without(previousSelections, deselectedRow);
    } else {
      this.selections = this.hasMultiSelect() ? [...previousSelections, toggledRow] : [toggledRow];
    }
    this.selectionsChange.emit(this.selections);
    this.changeDetector.markForCheck();
  }
```

Reply:
Yeah. This is much more cleaner. I got lazy when writing the previous code.

Reply:
done